### ISSUE TITLE
Message with Blocked Keyword Got Through

### ISSUE THREAD

**Issue by Astronox** [FossifyOrg/Messages#99]
### Checklist

- [X] I can reproduce the bug with the latest version given [here](https://github.com/FossifyOrg/Messages/releases/latest).
- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Messages/issues) or [closed](https://github.com/FossifyOrg/Messages/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Messages/discussions) or [closed](https://github.com/FossifyOrg/Messages/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Messages/blob/master/CONTRIBUTING.md).

### Affected app version

1.0.1

### Affected Android/Custom ROM version

GrapheneOS Uq1A.240205.004.2024020500

### Affected device model

Pixel 8

### How did you install the app?

F-Droid / IzzyOnDroid

### Steps to reproduce the bug

The text that should have been blocked was a political text. The keyword that should have been blocked was "stop=end". I suspect the message got through because it was sent with an image. 

Testing with friends revealed that:
-The original text only contents of the political text were blocked when sent by a friend.
-The SMS character limit is not a factor, the keyword was blocked when at the beginning of a short text and at the end of a long text.

Unfortunately, I forgot to ask them to send the text with an image, so I haven't tested MMS (and I'm a little hesitant to keep bothering my friends with this). Steps to reproduce are most likely:
1) Set a keyword to block (in this case "stop=end")
2) Have another party send an MMS with an image and the keyword
3) MMS should still be delivered with a text that should have been blocked.

### Expected behavior

Expected behavior is that any text message, SMS or MMS, image or not, should be blocked if it contains a keyword.
(As a side note, it would be nice to have the option to allow texts that do contain keywords through if they are sent by someone in your contacts).

### Actual behavior

A text message with a blocked keyword was received.

### Screenshots/Screen recordings

Evidence of blocked message (sent by a friend)
<img alt="ignoreImageMinify" src="https://github.com/FossifyOrg/Messages/assets/38494293/c85e40bc-bca3-4a47-a703-2541cf7ed6dd" width=359 />

Screenshot of message that was blocked (sent by a friend)
<img alt="ignoreImageMinify" src="https://github.com/FossifyOrg/Messages/assets/38494293/8ed4ea85-c27c-4148-b76a-45bf2a124172" width=291 />

Screenshot of the original text only version of the message being blocked, "Hello World" was sent after the original text message was sent. I sent them the original contents of the text, they just copied it and sent it back. It was blocked. (sent by a friend)
<img alt="ignoreImageMinify" src="https://github.com/FossifyOrg/Messages/assets/38494293/1039a97c-2a11-4d37-8057-1327241de495" width=464 />

Message failed to block (sent by political organization)
<img alt="ignoreImageMinify" src="https://github.com/FossifyOrg/Messages/assets/38494293/82c78e30-19a0-4cfa-9127-75b224cea475" width=255 />

### Additional information

_Edit by @naveensingh: triggered image minimizer_

**Comment by Aga-C** [comment]
I can confirm that MMS is not blocked.

**Comment by CaseOf** [comment]
phone numbers blocked are affected by this bug too

**Comment by naveensingh** [comment]
@CaseOf are you saying MMS from blocked phone numbers is not blocked?

**Comment by CaseOf** [comment]
> @CaseOf are you saying MMS from blocked phone numbers is not blocked?

on sms, I’m still getting notifications, but when opening the app, the message disappear.

**Comment by Astronox** [comment]
I understand this is a feature request, so it should probably get its own post, but would it be possible to allow your contacts to act as a whitelist? That way in case someone I know sends me a blocked keyword I can still see it.

**Comment by slhale** [comment]
I also have been getting this bug. I installed this messages app instead of my phone default because of the keyword blocking, but I constantly still get political messages like this despite them containing keywords that I have blocked. Also, I don't know how to check whether some are getting successfully blocked or not because I don't believe there is any indication in the app if I would have received a message but it was blocked.

**Comment by GrantOjanen** [comment]
I have the exact same problem. I tested it by having a family member send me both a SMS message and a MMS message, and I received the MMS, but the SMS was blocked.

Dang political scam texts have bothered me for about 5 years now. Reporting them to 7726 and replying STOP isn't really working.

**Comment by laubster** [comment]
Help! Blocked keywords has stopped working in 1.1.5, and I'm drowning in political texts. One example of many; it has an image & so is MMS:
Blocked keyword: stop2end
Final line of a text that was not blocked: -stop2end-
(that's dash stop2end dash)

Another example, which is MMS only due to length (no image), ends with: Stop2End

**Comment by Aga-C** [comment]
I've tested all previous versions, and it seems that this feature never worked correctly. The message at first is in fact blocked (no notification, nothing in conversation), but when the other message is received, the blocked one appears in the conversation.

**Comment by naveensingh** [comment]
> I've tested all previous versions, and it seems that this feature never worked correctly. The message at first is in fact blocked (no notification, nothing in conversation), but when the other message is received, the blocked one appears in the conversation.

@Aga-C Those were all MMS messages, right?

**Comment by Aga-C** [comment]
Yes

**Comment by laubster** [comment]
Leaks are still occurring in version 1.2.3; full text of latest appears below. One of the blocked keywords I have entered via Settings is "Stop to stop". (I couldn't find it mentioned in the UI, so a quick question: are those blocking keywords case sensitive? I'd have to have to enter "Stop To Stop", "Stop to Stop", "StOp tO sToP", etc. It' not applicable here, however: the keywords match exactly.)

What can I do to get this bug squashed? I'm happy to roll up my sleeves & do some work. (Or, make a $100 donation - yeah, I'm desperate.)  I've an extensive - though quite dated - background in multi-platform software compilation, but not on Android. If someone could point me to an idiots guide showing the steps for me to build my own copy & install it on my phone, I can litter some debug statements throughout the source & find the problem.

I vaguely recall mention of plans for a way to turn on debugging info generation, but I'm not yet seeing it in Settings. I'd be happy to add that too.

Thanks. :-)

The following was sent from "18337534841" , but they keep changing the number. There's an image is at the top, followed by this text:
Hi there. It's Gavin Newsom. I'm reaching out to respectfully ask you to make a contribution to help me pass the Election Rigging Response Act in California.

I want to stress the importance of this ask: ballots drop in just a few weeks and our opponents have already started paid advertising because one man with a history of giving to extremist causes and candidates contributed $10 million.

We will put your donation right to work getting on the air with our own ads. This is critical.

Folks... God help us if we're not successful here. We may have enjoyed our last free and fair election.

They're redrawing districts to hold the House -- talking to more and more Republican states; they're going after ActBlue and Democrats' ability to fundraise; going after colleges and the free press... We have to act.

So please - donate here if you can:

https://secure.actblue.com/donate/erra-txt?refcode=txt-250829b-supporters-we-need

Thank you. - Gavin
Paid for by www.stopelectionrigging.com. Top Funder: HMP. Stop to stop

**Comment by naveensingh** [comment]
Your ping gave me the push I needed. Time blocked on my calendar.

**Comment by antodc** [comment]
Just tried with my phone and the message was correctly blocked. Could it be that inadvertently you put an extra space somewhere in “Stop to stop”?

**Comment by naveensingh** [comment]
Was it an MMS?

**Comment by antodc** [comment]
> Was it an MMS?

No, only text. Just seen that @laubster mentioned that in his/her case there was an image. Perhaps that is causing the issue.

**Comment by naveensingh** [comment]
Yeah, it's a bug in the code. MMS messages are never filtered out properly, more details in [above comment](https://github.com/FossifyOrg/Messages/issues/99#issuecomment-2682724552).

**Comment by naveensingh** [comment]
I have a related question for everyone. Does **pattern-based number blocking** work properly when blocked senders send MMS messages? Asking because I can't test MMS myself.

**Comment by naveensingh** [comment]
A test APK is available here: https://github.com/FossifyOrg/Messages/actions/runs/18159066751/artifacts/4153075187

I have listed some test cases in https://github.com/FossifyOrg/Messages/pull/535. The fix is purely "theoretical" because I have never seen an MMS message in my life.

**Comment by laubster** [comment]
Thanks much, Naveen. I've installed it, added all the privs to make it the default Messenger app, exported/imported my blocked keywords list, and - with the assistance of my much hated work iphone - confirmed this debug app can both send & receive not only SMS but also MMS to a single individual with a photo. And, most happily, when I sent an image from my work phone with "Stop to end" as attached text, the whole message was suppressed. Encouraging!

The proof will be the political ads; the most recent arrived 2 days ago. I'll obviously be able to report if the fix doesn't help (a.k.a. spam slips through), but is there a way to give a definitive assessment that the fix is indeed working? (Some log that records datetime & reason for filtered messages?)

I just became a monthly sponsor. :-)

**Comment by naveensingh** [comment]
Thanks for testing.

> is there a way to give a definitive assessment that the fix is indeed working?

No, not at this moment.
